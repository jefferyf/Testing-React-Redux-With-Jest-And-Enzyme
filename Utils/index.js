import checkPropTypes from 'check-prop-types';
import { applyMiddleware, createStore } from 'redux';
import rootReducer from './../src/reducers/index';
import { middlewares } from './../src/createStore';

export const checkProps = (component, expectedProps) => {
  const propsError =  checkPropTypes(component.propTypes, expectedProps, 'props', component.name);
  return propsError;
};

export const findByTestAttr = (wrapper, attr) => {
  return wrapper.find(`[data-test='${attr}']`);
};

export const testStore = (initialState) => {
  const createStoreWithMiddleware = applyMiddleware(...middlewares)(createStore);
  return createStoreWithMiddleware(rootReducer, initialState);
}