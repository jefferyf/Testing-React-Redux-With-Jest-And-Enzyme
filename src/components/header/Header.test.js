import React from 'react';
import { shallow } from 'enzyme';
import { findByTestAttr } from '../../../Utils';
import { create } from 'react-test-renderer';
import Header from './Header';

const setUp = () => {
  const wrapper = shallow(<Header />);
  return wrapper;
}

describe('Header Component', () => {
  let wrapper;
  
  beforeEach(() => {
    wrapper = setUp();
  });
  
  describe('snapshot tests', () => {
    it('renders correctly', () => {
      const tree = create(wrapper);
  
      expect(tree.toJSON()).toMatchSnapshot();
    });

    // Personally, I think that snapshot tests are better for this
    // as it's kind of busywork, but these are defintely easier to 
    // reason about/read.
    describe('wrapper tests', () => {
      it('renders the logo correctly', () => {
        expect(findByTestAttr(wrapper, 'logoImg').length).toBe(1);
      });

      it('renders the headerComponent correctly', () => {
        expect(findByTestAttr(wrapper, 'headerComponent').length).toBe(1);
      });
    });
  });
});
