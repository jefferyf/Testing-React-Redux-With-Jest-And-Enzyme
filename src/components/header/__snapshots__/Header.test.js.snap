// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component snapshot tests renders correctly 1`] = `
<header
  data-test="headerComponent"
>
  <div
    className="wrap"
  >
    <div
      className="logo"
    >
      <img
        alt="logo"
        data-test="logoImg"
        src="logo.png"
      />
    </div>
  </div>
</header>
`;
